doc/build-netbsd.md

```diff
@@ -0,0 +1,49 @@
+NetBSD build guide
+======================
+(updated for NetBSD 7.0)
+
+This guide describes how to build bitcoind and command-line utilities on NetBSD.
+
+This guide does not contain instructions for building the GUI.
+
+Preparation
+-------------
+
+You will need the following modules, which can be installed via pkgsrc or pkgin:
+
+```
+autoconf
+automake
+boost
+db4
+git
+gmake
+libevent
+libtool
+python27
+```
+
+Download the source code:
+```
+git clone https://github.com/bitcoin/bitcoin
+```
+
+See [dependencies.md](dependencies.md) for a complete overview.
+
+### Building Bitcoin Core
+
+**Important**: Use `gmake` (the non-GNU `make` will exit with an error).
+
+With wallet:
+```
+./autogen.sh
+./configure CPPFLAGS="-I/usr/pkg/include -DOS_NETBSD" LDFLAGS="-L/usr/pkg/lib" BOOST_CPPFLAGS="-I/usr/pkg/include" BOOST_LDFLAGS="-L/usr/pkg/lib"
```

Comment:
It seems like we should be able to auto-detect /usr/pkg/*, then, no?

Reply:
If there's a way to include this in the Makefile (or whatever's most appropriate), then I certainly welcome such a change, especially if it simplifies the usage of "./configure" to something like this:  ./configure CPPFLAGS="-DOS_NETBSD"

The "/usr/pkg/" path is the default where pkgsrc and pkgin place package sources, binaries, libraries, configuration files, etc., and most (if not all) NetBSD installations have this.

Reply:
Yes, adding those flags seems unnecessary. Further, DOS_NETBSD can and should be set by configure.

Reply:
This is fantastic.  Eliminating the need for any flags with "./configure" is something I agree with, and placing it in the "configure.ac" file after "openbsd" seems reasonable to me (if, in the future, this list becomes longer, then at that point there may be a need to organize it in alphabetical order for easier maintenance, but for now it's short so I'm not suggesting this).

Reply:
@TheBlueMatt This is from the irc.freenode.net#netbsd channel regarding your question about auto-detecting the "/usr/pkg/" directory...

-----BEGIN-----
[22:07:50] <Riastradh> Randolf: Hmm...  Re your bitcoin pull request: software should _not_ look in /usr/pkg by default.  Anything built by pkgsrc will automatically add -I/usr/pkg/include -- or -I${PREFIX}/include, when PREFIX may not actually be /usr/pkg.
[22:08:39] <Riastradh> Part of the point of /usr/pkg is that nothing else uses it by default, unlike /usr/local which has all sorts of variegated cultural associations and random behaviour.
-----END-----

So, perhaps it would be better to not attempt to automatically detect NetBSD's pkgsrc/pkgin systems, but instead either leave the the ./configure parameters as documented with CPPFLAGS, etc., or provide an alternative such as a special parameter called PKG_PATH="/usr/pkg" that helps to make it all work automatically behind-the-scenes (assuming this is even possible).

Reply:
Better to not hardcode /usr/pkg, as this prefix is configurable and people use diverse ones, like /opt, ~/pkg/ etc.